### PR TITLE
Cleanup arrow_demo.

### PR DESCRIPTION
**Pull request by anntzer**
- Skip unnecessary numbered_bases_to_rates, lettered_bases_to_rates.
- Don't use pyplot, but pass axes to make_arrow_plot.
- letter colors only depend on the start point, not the end point.
- Don't hard-code coordinates of arrows, but compute them from the
  letter coordinates.
- Only include one dataset.
- Directly show all three possibilities (length, width, alpha).
- head_starts_at_zero=True resulted in incorrect alignment e.g. when
  calling `arrow_demo.py realistic length` previously; one should use
  the default value of head_starts_at_zero=False.

Before, when called with arguments `realistic length` (note the completely incorrect alignment):
old
After:
new

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: examples/text_labels_and_annotations/arrow_demo.py [hunk 1]
```diff
@@ -49,20 +38,14 @@ def make_arrow_plot(data, size=4, display='length', shape='right',
     alpha : float
         Maximum opacity of arrows.
     **kwargs
-        Can be anything allowed by a Arrow object, e.g. *linewidth* or
+        Can be anything allowed by an Arrow object, e.g. *linewidth* or
```
Comment (story645) [inline examples/text_labels_and_annotations/arrow_demo.py:41]: Is there anyway to cross link to the ArrowStyle object here?

File: examples/text_labels_and_annotations/arrow_demo.py [hunk 2]
```diff
@@ -235,73 +189,18 @@ def draw_arrow(pair, alpha=alpha, ec=ec, labelcolor=labelcolor):
...
-        scaled = False
-    if len(argv) > 2:
-        display = argv[2]
-    else:
-        display = 'length'
-
     size = 4
-    plt.figure(figsize=(size, size))
+    fig = plt.figure(figsize=(3 * size, size), constrained_layout=True)
+    axs = fig.subplot_mosaic([["length", "width", "alpha"]])
```
Comment (story645) [inline examples/text_labels_and_annotations/arrow_demo.py:199]: I find it really hard to learn these concepts from the demo (and I feel like @jklymak may also have opinions here). 

Is there a way to make it clearer, especially given how much code is in this demo? Either by making the different length, width, alpha values very dramatic, and/or adding the values to the axes label?

**Comment by anntzer** [comment]
Re: text size: actually they're all differently sized, but the difference is a bit subtle.  I increased the size of G to improve contrast (actually the sizes should probably be given by an eigenvector of the transition matrix, but let's not get there).

File: examples/text_labels_and_annotations/arrow_demo.py (hunk 1 above)
Comment (anntzer) [reply examples/text_labels_and_annotations/arrow_demo.py:41]: linked

File: examples/text_labels_and_annotations/arrow_demo.py (hunk 2 above)
Comment (anntzer) [reply examples/text_labels_and_annotations/arrow_demo.py:176]: That size is also used later to scale the text size, so I left it in as a variable.

**Comment by timhoffm** [comment]
> Directly show all three possibilities (length, width, alpha).

This is rather confusing without further context. I'm not convinced that this option is useful at all. It's definition and impelmentation are quite far apart. It complicates the code. And Having three different axes without any explanation is rather confusing.

I did not dig into the implementation, but I have the impression the author got carried away with making a fancy visualization, and the extra code obscures the actual demonstration of arrow usage. Possibly this should have been something much simpler in the first place.

I don't require to rewrite this whole thing, but as an incremental change

- either ditch the display variants alltogether
- or if you want to show all, write a sentence in the description that the different axes show certain variations in parameters and add titles "length modified", "width modified", "alpha modified" or similar.

**Comment by anntzer** [comment]
I extended the description at the axes corner to "flux encoded as arrow {length,width,alpha}".  I guess this may be a bit domain-specific but I thought that the example use case was actually quite clear.

**Comment by timhoffm** [comment]
> may be a bit domain-specific

Is a bit of an understatement. I did not have any clue that the arrows were fluxes. And I still don't have any idea what this diagram shows - which in itself is not a problem because all I want to learn is how to draw arrows. But the description doesn't really help much.

I have the impression that this example is more of the "show fancy plot" category similar to 
https://matplotlib.org/stable/gallery/lines_bars_and_markers/horizontal_barchart_distribution.html or https://matplotlib.org/stable/gallery/showcase/bachelors_degrees_by_gender.html
and not primarily an "Arrow Demo".

If this is some kind of standard domain specific plot, I suggest to rename it to "[domain specific plot name]" and add a description

"This example illustrates how one can encode flux densities in [domian specific plot name] using different arrow properties like, length, width or transparency."

File: examples/text_labels_and_annotations/arrow_demo.py [hunk 3]
```diff
@@ -5,151 +5,83 @@
 
 Arrow drawing example for the new fancy_arrow facilities.
```
Comment (jklymak) [inline examples/text_labels_and_annotations/arrow_demo.py:7]:
```suggestion
Three ways of drawing arrows to encode strength of the arrow using length or width of the arrow,
or alpha of its color.  
```

File: examples/text_labels_and_annotations/arrow_demo.py [hunk 4]
```diff
@@ -5,151 +5,83 @@
...
 
-Code contributed by: Rob Knight <rob@spot.colorado.edu>
-
-usage:
-
-  python arrow_demo.py realistic|full|sample|extreme
+Original author: Rob Knight <rob@spot.colorado.edu>
+"""
```
Comment (jklymak) [inline examples/text_labels_and_annotations/arrow_demo.py:9]:
```suggestion
"""
```

We don't cite people anymore..

**Review by jklymak — APPROVED** [review]
Aside from the comments about the front matter this is an improvement on the current situation in my opinion.  I agree that it should not get such a comprehensive name because it is really over the top for an arrow demo, and someone should devise a proper arrow demo, but until that time I personally think it should go in as more modern and more informative.

**Comment by anntzer** [comment]
I went mostly for @jklymak's suggestion, also mentioning that this can be used e.g. for Markov models (which is what I think this likely represented).  You can find Markov model graphs on google, although most of them just write the numbers without varying arrow width/height/alpha (but that's likely just for simpler drawing...).

**Comment by jklymak** [comment]
Anyone can merge after CI passes...  I'll open an issue tracking the concerns with this being overly specialized...

**Pull request merged**